### PR TITLE
Feature/es6 rewrite

### PR DESCRIPTION
**Pull request by dingo-d**
Rewrite the gulp using ES6 code, and added support for `gulp-babel` for ES6 support in JS files

**Needs testing!**

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2016"]
```
Comment (ahmadawais) [inline .babelrc]: I think presets `env` is better and the recommended practice as of now. 

Check https://babeljs.io/env/

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,277 @@
+{
```
Comment (ahmadawais) [inline .eslintrc]: This PR adds ESLint rules that I am not comfortable with. This is something we are discussing in the new version of WPGulp. You're more than welcomed to discuss it at WPGulp 2.0 Call For Contributors here → #61

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,18 +1,23 @@
 # Changelog
 
+### Version 1.0.4
+- NEW: Added .eslintrc and .babelrc
```
Comment (ahmadawais) [inline CHANGELOG.md]: Take a looks at the new [Contribution Guidelines for writing consistent commit messages](https://github.com/ahmadawais/WPGulp/blob/master/CONTRIBUTING.md) for this repo.

**Comment by ahmadawais** [comment]
Hey, @dingo-d!
Thanks for your contribution but I won't be able to accept it in this format. 

We have started [🚀 WPGulp 2.0 project](https://github.com/ahmadawais/WPGulp/issues/61) where you can take a look at how things are progressing.

Also, take a look at [💯 ES6 or ESNext Support #67](https://github.com/ahmadawais/WPGulp/issues/67) in case you want to take the lead there. 

ESLint configuration must be a separate PR as in there's a lot to consider here. The rules we should follow must be WordPress Coding Standards and I don't think that there are pre-written ESLint configs for that.

Looking forward!

**Comment by dingo-d** [comment]
Ok, so the rewrite should wait for the ESLint rules? I'll wait for that to be done, and I'll submit the PR when this is done. The general look of the `gulpfile.babel.js` will be the same (sans the spacing rules).

**Comment by ahmadawais** [comment]
Definitely, the ES6 port looks all good. Though, it doesn't work for
compiling Vendor JS which might be in ES6. Though it might be rare.

We also need help for ESLint + Prettier configuration of WPGulp. Would be
awesome if you take the lead on that?

Let me know.

**Comment by dingo-d** [comment]
I can lend a hand with it, although I would need to get in the Prettier, because I haven't used it before.

**Pull request closed without merging**